src/librustc_typeck/check/mod.rs

```diff
@@ -1041,6 +996,11 @@ fn check_impl_items_against_trait<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
         span_err!(tcx.sess, impl_span, E0046,
             "not all trait items implemented, missing: {}",
             missing_items.connect(", "));
+        if let Some(at) = overriden_associated_type {
+            span_note!(tcx.sess, at.span,
+                       "an overriden associated type \
+                        requires reimplementations of all trait-items");
```

Comment:
This error message could be much more helpful - showing what caused the problem and what needs to be fixed. Happy to see that as follow-up rather than doing it now, but please file an issue and add a FIXME here.